### PR TITLE
[MODFIN-291]-added manual validation for fiscalyearcode format.

### PR DESCRIPTION
**Pull request by SinghAdes**
## Purpose
https://issues.folio.org/browse/MODFIN-291

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by Abdulkhakimov** [comment]
Can you rename the PR title following by Thunderjet standard : [Jira ticket code] Jira ticket name?

**Comment by Abdulkhakimov** [comment]
Is it possible to cover with tests?

**Comment by SinghAdes** [comment]
> Can you rename the PR title following by Thunderjet standard : [Jira ticket code] Jira ticket name?

done.

**Comment by Abdulkhakimov** [comment]
@SinghAdes It

> > Can you rename the PR title following by Thunderjet standard : [Jira ticket code] Jira ticket name?
> 
> done.

Thank you

**Review by Abdulkhakimov — APPROVED** [review]

File: src/main/java/org/folio/rest/impl/FiscalYearsApi.java [hunk 1]
```diff
@@ -56,6 +61,10 @@ public void postFinanceFiscalYears(String lang, FiscalYear fiscalYear, Map<Strin
       .exceptionally(fail -> handleErrorResponse(handler, fail));
   }
 
+  private boolean isFiscalYearValid(FiscalYear fiscalYear) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/rest/impl/FiscalYearsApi.java:64]: We need the same valiation in PUT method

Comment (SinghAdes) [reply src/main/java/org/folio/rest/impl/FiscalYearsApi.java:64]: fixed

**Pull request merged**